src/core.c/ThreadPoolScheduler.pm6

```diff
@@ -772,13 +772,18 @@ my class ThreadPoolScheduler does Scheduler {
         }
     }
 
-    submethod BUILD(
-      Int() :$!initial_threads = 0,
-      Int() :$!max_threads =
-        %*ENV<RAKUDO_MAX_THREADS> // (Kernel.cpu-cores * 8 max 64)
-    --> Nil) {
+    submethod TWEAK(:$initial_threads, :$max_threads --> Nil) {
+        $!initial_threads = .Int with $initial_threads;
+        $!max_threads = nqp::istype($max_threads,Whatever)
+          ?? 9223372036854775807       # XXX should be -1
+          !! $max_threads.defined
+            ?? $max_threads == Inf
+              ?? 9223372036854775807   # XXX should be -1
+              !! $max_threads.Int
+            !! (%*ENV<RAKUDO_MAX_THREADS> // (Kernel.cpu-cores * 8 max 64)).Int;
```

Comment:
Also just a thought. Maybe set to "infinite" if the environment is -1 or just any negative? Could be helpful in detecting some problems too.